src/Google/Ads/GoogleAds/Util/FieldMasks.php

```diff
@@ -139,10 +158,6 @@ private static function buildPaths(
                             if (self::isWrapperType($fieldDescriptor->getMessageType())) {
                                 // For wrapper types, just emit the field name.
                                 $paths[] = $fieldName;
-                            } elseif (is_null($originalValue)) {
```

Comment:
I am not sure to understand how removing this recursion helps with the repeated field issue?

Reply:
This is not about repeated field only.
Originally, it's not quite necessary. It just go through all the fields of the message and try to add a new path to the field paths array. That's actually covered already by this method.

Besides, the `addNewFields` method is also buggy because it automatically adds the repeated field to the field paths.

Consider:

```php
$sitelinkFeedItem = new SitelinkFeedItem([
    'sitelink_text' => new StringValue(['value' => 'test']),
]);
```

Previously, it will not add only `sitelink_text` to the path, but also all repeated fields.

With the change in this PR, it will only add `sitelink_text`, which is more accurate.

Reply:
SGTM